### ISSUE TITLE
Unable to connect to Elasticsearch

### ISSUE THREAD

**Issue by TrickyBoy67** [deviantony/docker-elk#86]
Hello,

I don't know if it's a real issue or if I don't understand something but when I follow the steps, Kibana is `Unable to connect to Elasticsearch at http://elasticsearch:9200.`, so I don't know what to do.

What are the basics verifications to check if everything is running ok ?

Thanks in advance

**Comment by deviantony** [comment]
@TrickyBoy67 have a look at the logs.

`docker-compose logs`. Copy/paste the logs here.

**Comment by TrickyBoy67** [comment]
I've tried different configuration but non worked
I'm on macOS Sierra if it's help

`Attaching to dockerelk_kibana_1, dockerelk_logstash_1, dockerelk_elasticsearch_1
elasticsearch_1  | 2017-02-02T08:58:22.995926757Z [2017-02-02T08:58:22,994][INFO ][o.e.n.Node               ] [] initializing ...
kibana_1         | 2017-02-02T08:58:25.813980163Z {"type":"log","@timestamp":"2017-02-02T08:58:25Z","tags":["status","plugin:kibana@5.2.0","info"],"pid":11,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
elasticsearch_1  | 2017-02-02T08:58:23.163988010Z [2017-02-02T08:58:23,163][INFO ][o.e.e.NodeEnvironment    ] [_D4YuE3] using [1] data paths, mounts [[/usr/share/elasticsearch/data (/dev/sda1)]], net usable_space [55.1gb], net total_space [62.7gb], spins? [possibly], types [ext4]
kibana_1         | 2017-02-02T08:58:25.928248536Z {"type":"log","@timestamp":"2017-02-02T08:58:25Z","tags":["status","plugin:elasticsearch@5.2.0","info"],"pid":11,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
elasticsearch_1  | 2017-02-02T08:58:23.166922196Z [2017-02-02T08:58:23,166][INFO ][o.e.e.NodeEnvironment    ] [_D4YuE3] heap size [990.7mb], compressed ordinary object pointers [true]
kibana_1         | 2017-02-02T08:58:25.960663121Z {"type":"log","@timestamp":"2017-02-02T08:58:25Z","tags":["error","elasticsearch","admin"],"pid":11,"message":"Request error, retrying\nHEAD http://elasticsearch:9200/ => connect ECONNREFUSED 172.18.0.2:9200"}
kibana_1         | 2017-02-02T08:58:25.982628056Z {"type":"log","@timestamp":"2017-02-02T08:58:25Z","tags":["warning","elasticsearch","admin"],"pid":11,"message":"Unable to revive connection: http://elasticsearch:9200/"}
elasticsearch_1  | 2017-02-02T08:58:23.170213212Z [2017-02-02T08:58:23,169][INFO ][o.e.n.Node               ] node name [_D4YuE3] derived from node ID [_D4YuE3uTdCzzjSJBr-K8A]; set [node.name] to override
kibana_1         | 2017-02-02T08:58:25.983485059Z {"type":"log","@timestamp":"2017-02-02T08:58:25Z","tags":["warning","elasticsearch","admin"],"pid":11,"message":"No living connections"}
elasticsearch_1  | 2017-02-02T08:58:23.176841725Z [2017-02-02T08:58:23,176][INFO ][o.e.n.Node               ] version[5.2.0], pid[1], build[24e05b9/2017-01-24T19:52:35.800Z], OS[Linux/4.9.4-moby/amd64], JVM[Oracle Corporation/OpenJDK 64-Bit Server VM/1.8.0_111/25.111-b14]
kibana_1         | 2017-02-02T08:58:25.987757458Z {"type":"log","@timestamp":"2017-02-02T08:58:25Z","tags":["status","plugin:console@5.2.0","info"],"pid":11,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
elasticsearch_1  | 2017-02-02T08:58:24.498289241Z [2017-02-02T08:58:24,497][INFO ][o.e.p.PluginsService     ] [_D4YuE3] loaded module [aggs-matrix-stats]
kibana_1         | 2017-02-02T08:58:25.996082422Z {"type":"log","@timestamp":"2017-02-02T08:58:25Z","tags":["status","plugin:elasticsearch@5.2.0","error"],"pid":11,"state":"red","message":"Status changed from yellow to red - Unable to connect to Elasticsearch at http://elasticsearch:9200.","prevState":"yellow","prevMsg":"Waiting for Elasticsearch"}
elasticsearch_1  | 2017-02-02T08:58:24.498330536Z [2017-02-02T08:58:24,498][INFO ][o.e.p.PluginsService     ] [_D4YuE3] loaded module [ingest-common]
kibana_1         | 2017-02-02T08:58:26.374133822Z {"type":"log","@timestamp":"2017-02-02T08:58:26Z","tags":["status","plugin:timelion@5.2.0","info"],"pid":11,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
elasticsearch_1  | 2017-02-02T08:58:24.498334017Z [2017-02-02T08:58:24,498][INFO ][o.e.p.PluginsService     ] [_D4YuE3] loaded module [lang-expression]
kibana_1         | 2017-02-02T08:58:26.382348967Z {"type":"log","@timestamp":"2017-02-02T08:58:26Z","tags":["listening","info"],"pid":11,"message":"Server running at http://0.0.0.0:5601"}
elasticsearch_1  | 2017-02-02T08:58:24.498368080Z [2017-02-02T08:58:24,498][INFO ][o.e.p.PluginsService     ] [_D4YuE3] loaded module [lang-groovy]
kibana_1         | 2017-02-02T08:58:26.384048607Z {"type":"log","@timestamp":"2017-02-02T08:58:26Z","tags":["status","ui settings","error"],"pid":11,"state":"red","message":"Status changed from uninitialized to red - Elasticsearch plugin is red","prevState":"uninitialized","prevMsg":"uninitialized"}
elasticsearch_1  | 2017-02-02T08:58:24.498398697Z [2017-02-02T08:58:24,498][INFO ][o.e.p.PluginsService     ] [_D4YuE3] loaded module [lang-mustache]
kibana_1         | 2017-02-02T08:58:28.510236873Z {"type":"log","@timestamp":"2017-02-02T08:58:28Z","tags":["warning","elasticsearch","admin"],"pid":11,"message":"Unable to revive connection: http://elasticsearch:9200/"}
elasticsearch_1  | 2017-02-02T08:58:24.498546014Z [2017-02-02T08:58:24,498][INFO ][o.e.p.PluginsService     ] [_D4YuE3] loaded module [lang-painless]
kibana_1         | 2017-02-02T08:58:28.514220807Z {"type":"log","@timestamp":"2017-02-02T08:58:28Z","tags":["warning","elasticsearch","admin"],"pid":11,"message":"No living connections"}
elasticsearch_1  | 2017-02-02T08:58:24.498558362Z [2017-02-02T08:58:24,498][INFO ][o.e.p.PluginsService     ] [_D4YuE3] loaded module [percolator]
kibana_1         | 2017-02-02T08:58:31.187804809Z {"type":"log","@timestamp":"2017-02-02T08:58:31Z","tags":["warning","elasticsearch","admin"],"pid":11,"message":"Unable to revive connection: http://elasticsearch:9200/"}
elasticsearch_1  | 2017-02-02T08:58:24.498594023Z [2017-02-02T08:58:24,498][INFO ][o.e.p.PluginsService     ] [_D4YuE3] loaded module [reindex]
kibana_1         | 2017-02-02T08:58:31.442212002Z {"type":"log","@timestamp":"2017-02-02T08:58:31Z","tags":["warning","elasticsearch","admin"],"pid":11,"message":"No living connections"}
elasticsearch_1  | 2017-02-02T08:58:24.498816173Z [2017-02-02T08:58:24,498][INFO ][o.e.p.PluginsService     ] [_D4YuE3] loaded module [transport-netty3]
kibana_1         | 2017-02-02T08:58:38.420627915Z {"type":"log","@timestamp":"2017-02-02T08:58:38Z","tags":["status","plugin:elasticsearch@5.2.0","error"],"pid":11,"state":"red","message":"Status changed from red to red - Request Timeout after 3000ms","prevState":"red","prevMsg":"Unable to connect to Elasticsearch at http://elasticsearch:9200."}
elasticsearch_1  | 2017-02-02T08:58:24.498831891Z [2017-02-02T08:58:24,498][INFO ][o.e.p.PluginsService     ] [_D4YuE3] loaded module [transport-netty4]
elasticsearch_1  | 2017-02-02T08:58:24.499534379Z [2017-02-02T08:58:24,499][INFO ][o.e.p.PluginsService     ] [_D4YuE3] no plugins loaded
elasticsearch_1  | 2017-02-02T08:58:24.899689924Z [2017-02-02T08:58:24,899][WARN ][o.e.d.s.g.GroovyScriptEngineService] [groovy] scripts are deprecated, use [painless] scripts instead
elasticsearch_1  | 2017-02-02T08:58:28.160344391Z [2017-02-02T08:58:28,160][INFO ][o.e.n.Node               ] initialized
elasticsearch_1  | 2017-02-02T08:58:28.161126204Z [2017-02-02T08:58:28,160][INFO ][o.e.n.Node               ] [_D4YuE3] starting ...
elasticsearch_1  | 2017-02-02T08:58:28.340081520Z [2017-02-02T08:58:28,339][WARN ][i.n.u.i.MacAddressUtil   ] Failed to find a usable hardware address from the network interfaces; using random bytes: 34:c9:54:55:a1:56:a8:00
elasticsearch_1  | 2017-02-02T08:58:28.429933370Z [2017-02-02T08:58:28,428][INFO ][o.e.t.TransportService   ] [_D4YuE3] publish_address {172.18.0.2:9300}, bound_addresses {[::]:9300}
elasticsearch_1  | 2017-02-02T08:58:28.462786153Z [2017-02-02T08:58:28,462][INFO ][o.e.b.BootstrapChecks    ] [_D4YuE3] bound or publishing to a non-loopback or non-link-local address, enforcing bootstrap checks
elasticsearch_1  | 2017-02-02T08:58:31.720686736Z [2017-02-02T08:58:31,719][INFO ][o.e.c.s.ClusterService   ] [_D4YuE3] new_master {_D4YuE3}{_D4YuE3uTdCzzjSJBr-K8A}{zB1nfbsARMaCcuth9hXpVw}{172.18.0.2}{172.18.0.2:9300}, reason: zen-disco-elected-as-master ([0] nodes joined)
elasticsearch_1  | 2017-02-02T08:58:31.809942723Z [2017-02-02T08:58:31,809][INFO ][o.e.h.HttpServer         ] [_D4YuE3] publish_address {172.18.0.2:9200}, bound_addresses {[::]:9200}
elasticsearch_1  | 2017-02-02T08:58:31.809979235Z [2017-02-02T08:58:31,809][INFO ][o.e.n.Node               ] [_D4YuE3] started
elasticsearch_1  | 2017-02-02T08:58:31.906756952Z [2017-02-02T08:58:31,906][INFO ][o.e.g.GatewayService     ] [_D4YuE3] recovered [0] indices into cluster_state`

**Comment by deviantony** [comment]
Have you tried to access `localhost:5601` ? This error message can be displayed while Elasticsearch is starting. Kibana should automatically reconnect when Elasticsearch is ready.

Also what's the output of `docker-compose ps` after that?

**Comment by nguse** [comment]
I had this same issue.

Note the error line: Failed to find a usable hardware address from the network interfaces; using random bytes:


I was able to get this working again by using:
FROM elasticsearch:5.2.0

**Comment by deviantony** [comment]
@nguse strange, `elasticsearch:5` (as used in the Dockerfile) is pointing to `elasticsearch:5.2.0`. Maybe you just needed to pull the latest images using `docker-compose pull` ?

**Comment by TrickyBoy67** [comment]
It's when I'm accessing Kibana that I've this error, 

<img width="1208" alt="capture d ecran 2017-02-02 a 22 20 44" src="https://cloud.githubusercontent.com/assets/10465322/22569147/b1824e60-e996-11e6-99bf-10cec604aad0.png">

and docker-compose ps 

<img width="819" alt="dockercomposeps" src="https://cloud.githubusercontent.com/assets/10465322/22567100/dcc52bd6-e98e-11e6-8f2a-08182531ea6a.png">

**Comment by deviantony** [comment]
Are you using the provided `docker-compose.yml` file ? Or have you modified it?

**Comment by TrickyBoy67** [comment]
Yes I was using the provided file.

I've tried @nguse proposition but same result 

I've looked in the docker elasticsearch repo and it's seem that the 5.2.0 is 2 days old

**Comment by deviantony** [comment]
This is strange as I've just started the stack from scratch without any issue in 5.2.

Could you reset the stack and try again with `docker-compose down && docker-compose up` ?

**Comment by TrickyBoy67** [comment]
I have try to delete all images/containers, re try everything, so I've created a linux VM and I'm working in it and it's working. 

For the moment I'll stay with this solution 
Thanks for the help

**Comment by nguse** [comment]
This is really strange, making that change and then pulling/building did fix it for me, but I changed it back, built/pulled images again, and it continued to work.

This machine I put it on was brand new, I first installed Docker and tried to start this stack about an hour before I posted yesterday.

Seeing we both had the same issue for a short period of time and it was resolved almost right away, I'm going to just assume it was some temporary upstream issue and has been resolved.
